mk/spksrc.cross-rust.mk

```diff
@@ -56,6 +56,7 @@ CARGO_INSTALL_ARGS += --root $(STAGING_INSTALL_PREFIX)
 # Default build with rust and installation with cargo
 rust_install_target:
 	@echo "  ==> Cargo install rust package $(PKG_NAME)"
+	@$(RUN) rustup target install $(RUST_TARGET)
```

Comment:
@th0ma7 without this change, only x64 target builds do not fail. (local and github builds).
is there a better solution for this?

Reply:
This is odd... Would you hapen to be running on an "unclean" tree?  I just tried it from a clean tree within `cross` and it worked out ok without any need of patching.

Try cleaning-up your toolchain directory or simply delete the rust status file from there.  Normally that should do it.

Reply:
> This is odd... Would you hapen to be running on an "unclean" tree? I just tried it from a clean tree within `cross` and it worked out ok without any need of patching.
> 
> Try cleaning-up your toolchain directory or simply delete the rust status file from there. Normally that should do it.

It failed also on github build action that does the delete of rust status now.

Reply:
This is really really weird.  Yesterday I added `dssim` to `gstreamer` PR https://github.com/SynoCommunity/spksrc/pull/5298 which is rust based (`spksrc.cross-rust.mk`).  I was expecting the PR to fail similarly to yours but it didn't.  Everything worked out just fine.

Could it be possible that your PR's build image file is polluted somehow?  Could a rebase to master solve things?

Reply:
@th0ma7 just rebased this PR and removed the line above. Still the same. Only x64 package build without the additional rustup target install ... line.
So I have added it again.

Reply:
I may have found hints of what`s going on, using a clean image and tree I was able to reproduce your issue.

When digging further I noticed the following:
```
spksrc@spksrc:~/hgy59$ find . -name cargo | grep bin
./spksrc/distrib/cargo/bin/cargo
./spksrc/distrib/rustup/toolchains/stable-x86_64-unknown-linux-gnu/bin/cargo
./spksrc/distrib/rustup/toolchains/1.66.0-x86_64-unknown-linux-gnu/bin/cargo

spksrc@spksrc:~/hgy59$ ll ./spksrc/distrib/rustup/toolchains/stable-x86_64-unknown-linux-gnu/bin/cargo ./spksrc/distrib/rustup/toolchains/1.66.0-x86_64-unknown-linux-gnu/bin/cargo
-rwxr-xr-x 1 spksrc spksrc 25722120 Feb 25 16:07 ./spksrc/distrib/rustup/toolchains/1.66.0-x86_64-unknown-linux-gnu/bin/cargo
-rwxr-xr-x 1 spksrc spksrc 26054568 Feb 25 15:55 ./spksrc/distrib/rustup/toolchains/stable-x86_64-unknown-linux-gnu/bin/cargo
```
For some odd reason `cargo` switches from "stable" install at toolchain time to an actual version being `1.66.0` at build time ...  still investigating... but getting closer.

Reply:
The following looks like solving the version changing issue, can you test that out?
```
$ git diff .
diff --git a/mk/spksrc.cross-rust-env.mk b/mk/spksrc.cross-rust-env.mk
index f398432e..a3a7d14c 100644
--- a/mk/spksrc.cross-rust-env.mk
+++ b/mk/spksrc.cross-rust-env.mk
@@ -8,7 +8,9 @@ export RUSTUP_HOME=$(BASE_DISTRIB_DIR)/rustup
 export PATH:=$(BASE_DISTRIB_DIR)/cargo/bin:$(PATH)
 endif
 
-RUST_TOOLCHAIN ?= stable
+ifeq ($(RUST_TOOLCHAIN),)
+RUST_TOOLCHAIN = stable
+endif
 
 RUST_TARGET =
 # map archs to rust targets
diff --git a/mk/spksrc.cross-rust.mk b/mk/spksrc.cross-rust.mk
index 534f708c..d76fa1bc 100644
--- a/mk/spksrc.cross-rust.mk
+++ b/mk/spksrc.cross-rust.mk
@@ -56,8 +56,7 @@ CARGO_INSTALL_ARGS += --root $(STAGING_INSTALL_PREFIX)
 # Default build with rust and installation with cargo
 rust_install_target:
        @echo "  ==> Cargo install rust package $(PKG_NAME)"
-       @$(RUN) rustup target install $(RUST_TARGET)
-       @$(RUN) cargo install $(CARGO_INSTALL_ARGS)
+       @$(RUN) cargo +$(RUST_TOOLCHAIN) install $(CARGO_INSTALL_ARGS)
 
 
```